content/en/docs/v3.6/op-guide/security.md

```diff
@@ -239,6 +239,105 @@ To disable certificate chain checking, invoke curl with the `-k` flag:
 $ curl -k https://127.0.0.1:2379/v2/keys/foo -Xput -d value=bar -v
 ```
 
+## Example 5: Convert an existing non-TLS etcd cluster to TLS
+
+Previously, we described that etcd supports two types of communication: **client-to-server** and **server-to-server (peer communication)**. In this section, we will prepare configuration steps to migrate both communication types to TLS in your existing etcd cluster.
+
+> **Important**: Before restarting any etcd members, you must first add the TLS certificates and configuration to each etcd node.
+
+---
+
+### 1. Check current cluster status
+
+```sh
+export ETCDCTL_API=3
+
+# Define etcd node IPs
+HOST_1=ip_address_etcd_node_1
+HOST_2=ip_address_etcd_node_2
+HOST_3=ip_address_etcd_node_3
+
+# Get member list
+etcdctl --endpoints=${HOST_1}:2379,${HOST_2}:2379,${HOST_3}:2379 member list
+```
+
+You will get an output similar to:
+
+```sh
+10f4390f4a904a13, started, etcd-node-1, http://ip_address_etcd_node_1:2380, http://ip_address_etcd_node_1:2379, false
+18e03f3590bf045c, started, etcd-node-2, http://ip_address_etcd_node_2:2380, http://ip_address_etcd_node_2:2379, false
+ca7a7d6fe0984fd9, started, etcd-node-3, http://ip_address_etcd_node_3:2380, http://ip_address_etcd_node_3:2379, false
+```
+
+> Save the **member IDs** and corresponding **IP addresses** — you'll need them to update peer URLs.
+
+---
+
+### 2. Add TLS certs on each node
+
+Please add TLS certificates to each node using the instructions above.
+
+### 3. Enable TLS for client-to-server communication
+
+Update each node's parameters to match your etcd deployment method using the following flags:
+
+```ini
+--cert-file=/etc/ssl/etcd/server.pem
+--key-file=/etc/ssl/etcd/server-key.pem
+--client-cert-auth=true
+--trusted-ca-file=/etc/ssl/etcd/ca.pem
+--peer-cert-file=/etc/ssl/etcd/peer.pem
+--peer-key-file=/etc/ssl/etcd/peer-key.pem
+--peer-client-cert-auth=true
+--peer-trusted-ca-file=/etc/ssl/etcd/ca.pem
+```
+
+Also, update all `--listen-peer-urls`, `--listen-client-urls`, `--advertise-client-urls`, `--initial-cluster`  and `--initial-advertise-peer-urls` to use `https://`.
```

Comment:
After step 3 is done, it would be better to restart all etcd instances, and verify the client-server communication is working as expected before users go ahead to continue to update peer-to-peer communication.

Reply:
But if we reboot the nodes before doing member update etcd the node will hang. I checked this earlier.

Reply:
Do I understand correctly that you propose to split the step into two?
First, update the connection between etcd nodes (what you specified is responsible for mTLS between nodes) and only after that update the connection to TLS for the peer?

Below I have a comment about updating urls from http to https, I can move it to the main block where the TLS parameters are.